src/fides/api/service/connectors/dynamic_erasure_email_connector.py

```diff
@@ -0,0 +1,258 @@
+        split_field = dataset_reference.field.split(".")
+        collection_name = split_field[0]
+        collection_address = CollectionAddress(graph_dataset.name, collection_name)
+        # And get the field name from the datasetreference
+        field_name = ".".join(split_field[1:])
+
+        # Get the corresponding ConnectionConfig instance
+        connection_config: ConnectionConfig = (
+            db.execute(
+                db.query(ConnectionConfig).filter(
+                    ConnectionConfig.id == dataset_config.connection_config_id
+                )
+            )
+            .scalars()
+            .first()
+        )
+
+        # Instatiate the ConnectionConfig's connector so that we can execute the query
+        # to retrieve the email addresses based on the custom request fields
+        connector = get_connector(connection_config)
+
+        skipped_privacy_requests: List[str] = []
+        # We'll batch identities for each email address
+        # so we'll only send 1 email to each email address
+        batched_identities: Dict[str, List[str]] = {}
+
+        for privacy_request in privacy_requests:
+            email = self.get_email_address_from_custom_request_fields(
+                connector,
+                graph_dataset,
+                privacy_request,
+                collection_address,
+                field_name,
+            )
+
+            user_identities: Dict[str, Any] = privacy_request.get_cached_identity_data()
+            filtered_user_identities: Dict[str, Any] = (
+                filter_user_identities_for_connector(self.config, user_identities)
+            )
+
+            # If we couldn't get the email or the user identities, we skip the privacy request
+            if not email or not filtered_user_identities:
+                skipped_privacy_requests.append(privacy_request.id)
+                self.add_skipped_log(db, privacy_request)
+                continue
+
+            if email not in batched_identities:
+                batched_identities[email] = []
+
+            batched_identities[email].extend(filtered_user_identities.values())
+
+        if not batched_identities:
+            logger.info(
+                "Skipping erasure email send for connector: '{}'. "
+                "No corresponding user identities or email addresses found for pending privacy requests.",
+                self.configuration.name,
+            )
+            return
+
+        logger.info(
+            "Sending batched erasure email for connector {}...",
+            self.configuration.name,
+        )
+
+        for email_address, identities in batched_identities.items():
+            try:
+                send_single_erasure_email(
+                    db=db,
+                    subject_email=email_address,
+                    subject_name=self.config.third_party_vendor_name,
+                    batch_identities=identities,
+                    test_mode=False,
+                )
```

Comment:
Instead of sending one single erasure email, we send one email per recipient email address